### PR TITLE
Keystone connector documentation

### LINKED ISSUE #89: Keystone connector documentation
In dex [v2.14.0](https://github.com/dexidp/dex/releases/tag/v2.14.0), keystone connector was added. 
It still works, and it even has integration tests, but sadly, the keystone connector has no documentation.

### PR DESCRIPTION
**Pull request by nabokihms**
Signed-off-by: m.nabokikh <maksim.nabokikh@flant.com>

Fixes #89

### CODE REVIEW

File: content/docs/connectors/keystone.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+---
+title: "Authentication Through OpenStack Keystone"
+linkTitle: "OpenStack Keystone"
+description: ""
+date: 2020-09-30
+draft: false
+toc: true
+weight: 2100
```
Comment (nate-double-u) [inline content/docs/connectors/keystone.md:8]: This has the same weight as Bitbucket Cloud, so they'll default to alphabetical ordering between them.

If we want Keystone between Bitbucket cloud and Openshift, I'd suggest a weight between the two:

```suggestion
weight: 2105
```

File: content/docs/connectors/keystone.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+---
+title: "Authentication Through OpenStack Keystone"
+linkTitle: "OpenStack Keystone"
+description: ""
+date: 2020-09-30
```
Comment (nate-double-u) [inline content/docs/connectors/keystone.md:5]: This date seems incorrect -- v2.14.0 was released on Jan 11, 2019, but for documentation we may want to use the day the page was published or written.

Comment (nabokihms) [reply content/docs/connectors/keystone.md:5]: Overlooked it, thank you. Need time to get used to it.

File: content/docs/connectors/keystone.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+draft: false
+toc: true
+weight: 2100
+---
+
+## Overview
+
+[Keystone](https://docs.openstack.org/keystone/latest/) is an OpenStack service that provides API client authentication, service discovery, and distributed multi-tenant authorization. 
+
+OpenStack Keystone connector supports `offline_access` and `groups` scopes. To use this connector, you need to create a domain and user with an admin role and then specify credentials in the configuration file like in the example below.
```
Comment (nate-double-u) [inline content/docs/connectors/keystone.md:15]:
```suggestion
OpenStack Keystone connector supports `offline_access` and `groups` scopes. To use this connector, create a domain and user with an admin role, then specify the credentials in the configuration file (see the example below).
```

File: content/docs/connectors/keystone.md (hunk 1 above)
Comment (nabokihms) [reply content/docs/connectors/keystone.md:8]: I think the right place is after Gitea connector, so `2135`.

File: content/docs/connectors/keystone.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+Keystone should expose [Identity API v3](https://docs.openstack.org/api-ref/identity/v3/) to work with dex.
```
Comment (nate-double-u) [inline content/docs/connectors/keystone.md:16]: The term "should" is throwing me off a little here. Makes me think that sometimes Keystone doesn't expose the API... If that's the case, should we add a workaround?

```suggestion
Keystone exposes the [Identity API v3](https://docs.openstack.org/api-ref/identity/v3/) to work with dex.
```

File: content/docs/connectors/keystone.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+
+## Configuration
+
+The following is an example of a configuration for dex config file:
```
Comment (nate-double-u) [inline content/docs/connectors/keystone.md:21]: I don't do technical reviews for this project (yet -- still learning!), so i have a question about which file this is referring to -- in other connector examples it's referring to `examples/config-dev.yaml`, but I'm not 100% that's right either. 

The other option may be to remove this line as a little redundant (some of the other connectors don't have a blurb in their `## Configuration` section, they just go straight into the config.

```suggestion
The following is an example of an OpenStack Keystone configuration for dex:
```

**Comment by nate-double-u** [comment]
Thanks for this @nabokihms! I've done a copy-edit pass, but I tend not to do technical reviews for dex (yet 🙂). I've invited @sagikazarmark to look as well.

File: content/docs/connectors/keystone.md (hunk 5 above)
Comment (nabokihms) [reply content/docs/connectors/keystone.md:21]: As far as I remember, we want to abstain from using `examples/config-dev.yaml`.
https://github.com/dexidp/dex/blob/master/examples/config-dev.yaml#L1

But I don't think it is a good idea to point users to another configuration file  🤔 
I think I will apply your suggestion for now.

**Comment by nate-double-u** [comment]
Looks good to me, thanks @nabokihms

File: content/docs/connectors/keystone.md (hunk 5 above)
Comment (nate-double-u) [reply content/docs/connectors/keystone.md]: > As far as I remember, we want to abstain from using examples/config-dev.yaml.

I hadn't realized that--good to know.

**Review by sagikazarmark — APPROVED** [review]

**Pull request merged**